### PR TITLE
Feature: Add Subgraph step-by-step guide

### PR DESCRIPTION
**Pull request by josemarinas**

### CODE REVIEW

File: packages/subgraph/GUIDE.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+- `SUBGRAPH_VERSION`: The version of the subgraph.
+- `SUBGRAPH_NETWORK_NAME`: The network where the subgraph will be deployed. This will be used for generating the subgraph.yaml file.
+
+Editing this variables you can change how the subgraph is built and deployed.
+
+## Creating a Subgraph
+
+### `manifest/subgraph.placeholder.yaml`
+
+The first step to create a subgraph is to create the `subgraph.placeholder.yaml` file. This file is located in the `subgraph` folder. This file contains the template for the `subgraph.yaml` file. This file is used by the `build-manifest.sh` script to generate the `subgraph.yaml` file. You can find more information about the `subgraph.yaml` file [here](https://thegraph.com/docs/define-a-subgraph#the-subgraph-manifest).
```
Comment (Rekard0) [inline packages/subgraph/GUIDE.md]:
```suggestion
The first step to create a subgraph is to create or edit the `subgraph.placeholder.yaml` file. This file is located in the `manifest` folder. This file contains the template for the `subgraph.yaml` file. This file is used by the `build-manifest.sh` script to generate the `subgraph.yaml` file. You can find more information about the `subgraph.yaml` file [here](https://thegraph.com/docs/define-a-subgraph#the-subgraph-manifest).
```

File: packages/subgraph/GUIDE.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+You will see that the `subgraph.placeholder.yaml` file contains some placeholders that will be replaced by the `build-manifest.sh` script. These placeholders are the following:
+
+- `{{dataSources.PluginSetupProcessors}}`: The object containing the processors for the plugin setup events.
+- `{{startBlock}}`: The block number where the subgraph will start indexing.
+- `{{address}}`: The address of the contract that will be indexed.
+- `{{network}}`: The network where the contract is deployed.
+
+This placeholder are substituted using moustache. You can find more information about it [here](https://mustache.github.io/).
+
+The files containing the replacement values are stored in `manifest/data`. Here you can find a JSON file for each network. The name of the file is the name of the network. For example, the file for the mainnet network is `mainnet.json`. This file contains a property called `dataSources` that the information about the contract that will be indexed. In this case, we are only indexing one contract but you can index multiple contracts by adding more objects to the `dataSources` object.
```
Comment (Rekard0) [inline packages/subgraph/GUIDE.md]:
```suggestion
The files containing the replacement values are stored in `manifest/data`. Here you can find a JSON file for each network. The name of the file is the name of the network. For example, the file for the mainnet network is `mainnet.json`. This file contains a property called `dataSources`, which holds the information about the contract that will be indexed. In this case, we are only indexing one contract but you can index multiple contracts by adding more objects to the `dataSources` object.
```

File: packages/subgraph/GUIDE.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+### `schema.graphql`
+
+The first step to create a subgraph is to define the schema. The schema is defined in the `schema.graphql` file located in the `subgraph` folder. This file contains the GraphQL schema that will be used for querying the subgraph. The schema is defined using the GraphQL Schema Definition Language (SDL). You can find more information about it [here](https://graphql.org/learn/schema/).
```
Comment (Rekard0) [inline packages/subgraph/GUIDE.md]:
```suggestion
The second step to create a subgraph is to define the schema. The schema is defined in the `schema.graphql` file located in the `subgraph` folder. This file contains the GraphQL schema that will be used for querying the subgraph. The schema is defined using the GraphQL Schema Definition Language (SDL). You can find more information about it [here](https://graphql.org/learn/schema/).
```

File: packages/subgraph/GUIDE.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+```bash
+# SUBGRAPH
+
+## The Graph credentials
+GRAPH_KEY="zzzzzzzzzzzz"
+
+## Subgraph
+SUBGRAPH_NAME="osx"
+SUBGRAPH_VERSION="v1.0.0"
+SUBGRAPH_NETWORK_NAME="mainnet" #  ["mainnet", "goerli", "sepolia", "polygon", "polygonMumbai", "base", "baseGoerli", "arbitrum", "arbitrumGoerli"]
```
Comment (mathewmeconry) [inline packages/subgraph/GUIDE.md]: replace `arbitrumGoerli` and `baseGoerli` with the sepolia replacements

**Review by mathewmeconry — DISMISSED** [review]

File: USAGE_GUIDE.md [hunk 5]
```diff
@@ -35,11 +74,11 @@
 
    If you deploy upcoming versions of your plugin, you must increment the build or release number accordingly (see [our docs on versioning your plugin](https://devs.aragon.org/docs/osx/how-to-guides/plugin-development/publication/versioning)).
 
-### Testing
+### Contract Testing
```
Comment (heueristik) [inline USAGE_GUIDE.md]: Please create a `Contract` heading as it was done before and do not repeat the term `Contract` in every sub-heading.

File: USAGE_GUIDE.md [hunk 6]
```diff
@@ -76,20 +115,96 @@ The standard deploy scripts in the `packages/contracts/deploy` should already be
...
+After that you can install the dependencies by running:
+
+```bash
+cd packages/subgraph/
+```
+
+```bash
+yarn install
+```
+
```
Comment (heueristik) [inline USAGE_GUIDE.md]: This part duplicates the README.md. Let's not introduce more duplication. Take a look what already exists and reference existing texts.

**Review by heueristik — CHANGES_REQUESTED** [review]
Please reduce duplication and use a spell-checker (e.g. https://marketplace.visualstudio.com/items?itemName=adamvoss.vscode-languagetool)

File: USAGE_GUIDE.md [hunk 7]
```diff
@@ -76,20 +115,96 @@ The standard deploy scripts in the `packages/contracts/deploy` should already be
...
+The second step to create a subgraph is to define the schema. The schema is defined in the `schema.graphql` file located in the `subgraph` folder. This file contains the GraphQL schema that will be used for querying the subgraph. The schema is defined using the GraphQL Schema Definition Language (SDL). You can find more information about it [here](https://graphql.org/learn/schema/).
 
-Finally, the events and event handlers must then be linked together in the `subgraph.placeholder.yaml` inside the `packages/subgraph/manifest` folder.
+Here you can add your custom entities and how they are related to each other. You can find more information about the schema [here](https://thegraph.com/docs/define-a-subgraph#defining-the-schema).
 
-### Testing
+## Handlers
 
-The `packages/subgraph/test` folder contains a `plugin` and `osx` subfolder where the tests for the event handlers from the previous section are written.
+The handling functions will be handling the events specified in the `subgraph.yaml` file. The functions are located in the `src` folder. For this specific case we have 2 handlers:
```
Comment (heueristik) [inline USAGE_GUIDE.md]:
```suggestion
Events handler functions are associated with the events in the `subgraph.yaml` file and defined in the `src` folder. For the `MyPlugin` example, two handlers are already provided:
```

File: USAGE_GUIDE.md [hunk 8]
```diff
@@ -76,20 +115,96 @@ The standard deploy scripts in the `packages/contracts/deploy` should already be
...
 
-In `plugint.ts` inside the `plugin` folder, you can test the plugin-related event handlers.
-In `pluginSetupProcessor.ts` inside the `osx` folder, you can test plugin-setup-processor-related event handlers that your plugin might require.
+- `handleInstallationPrepared`: Handeles the `InstallationPrepared` event. This event is emitted when the installation is prepared. This event contains the `installationId` and the handler will create a new `PluginEntity` entity with the `installationId` as the id.
```
Comment (heueristik) [inline USAGE_GUIDE.md]:
```suggestion
- `handleInstallationPrepared`: Handles the `InstallationPrepared` event. This event is emitted when the installation is prepared. This event contains the `installationId` and the handler will create a new `PluginEntity` entity with the `installationId` as the id.
```

Is all the information provided here necessary to use the template? Just listing the event fields doesn't add much value to the Guide, IMO.

Please go throught the text once more with this mindset.

File: USAGE_GUIDE.md (hunk 5 above)
Comment (Rekard0) [reply USAGE_GUIDE.md]: addressed in [6e1e9e1](https://github.com/aragon/osx-plugin-template-hardhat/pull/25/commits/6e1e9e1354f20ae7c695e0ea31dd55841e73ca5f)

**Comment by Rekard0** [comment]
> Please reduce duplication and use a spell-checker (e.g. https://marketplace.visualstudio.com/items?itemName=adamvoss.vscode-languagetool)

@josemarinas FYI

File: USAGE_GUIDE.md [hunk 9]
```diff
@@ -76,20 +102,91 @@ The standard deploy scripts in the `packages/contracts/deploy` should already be
...
+You will see that the `subgraph.placeholder.yaml` file contains some placeholders that will be replaced by the `build-manifest.sh` script. These placeholders are the following:
 
-### Handling Events
+- `{{dataSources.PluginSetupProcessors}}`: The object containing the processors for the plugin setup events.
+- `{{startBlock}}`: The block number where the subgraph will start indexing.
+- `{{address}}`: The address of the contract that will be indexed.
+- `{{network}}`: The network where the contract is deployed.
 
-The `packages/subgraph/src` folder contains a `plugin` and `osx` subfolder.
+This placeholder are substituted using moustache. You can find more information about it [here](https://mustache.github.io/).
```
Comment (heueristik) [inline USAGE_GUIDE.md]:
```suggestion
These placeholders are substituted using [mustache](https://mustache.github.io/).
```

File: USAGE_GUIDE.md [hunk 10]
```diff
@@ -76,20 +102,91 @@ The standard deploy scripts in the `packages/contracts/deploy` should already be
...
 
-### Testing
+- `handleInstallationPrepared`: Handles the `InstallationPrepared` event. This event is emitted when the installation is prepared. This event contains the `installationId` and the handler will create a new `PluginEntity` entity with the `installationId` as the id.
+- `handleNumberStored`: Handles the `NumberStored` event. And stores the number in the `pluginEntity` entity in the `number` property.
+
+## Testing
 
-The `packages/subgraph/test` folder contains a `plugin` and `osx` subfolder where the tests for the event handlers from the previous section are written.
+This template uses [matchstick-as](https://github.com/LimeChain/matchstick) framework for unit testing.
+Similar to the `src` folder, the `test` folder contains a `plugin` and `osx` subfolder where the tests for the event handlers from the previous section are written.
```
Comment (heueristik) [inline USAGE_GUIDE.md]:
```suggestion
Similar to the `src` folder, the `test` folder contains a `plugin` and `osx` subfolder, where the tests for the event handlers from the previous section are written.
```

**Review by heueristik — APPROVED** [review]

**Pull request merged**